compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -386,14 +400,13 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
                 }
             }
 
-            let (lifetimes, binders): (FxIndexMap<LocalDefId, Region>, Vec<_>) =
+            let (lifetimes, binders): (FxIndexMap<LocalDefId, ResolvedArg>, Vec<_>) =
                 bound_generic_params
                     .iter()
-                    .filter(|param| matches!(param.kind, GenericParamKind::Lifetime { .. }))
                     .enumerate()
                     .map(|(late_bound_idx, param)| {
-                        let pair = Region::late(late_bound_idx as u32, param);
-                        let r = late_region_as_bound_region(self.tcx, &pair.1);
+                        let pair = ResolvedArg::late(late_bound_idx as u32, param);
+                        let r = late_arg_as_bound_arg(self.tcx, &pair.1, param);
                         (pair, r)
                     })
                     .unzip();
```

Comment:
Pre-existing: this would be clearer split into 2 loops, one running on the other's result.

Reply:
Problem here is that both loops now need access to `param`, so the second loop would need to `zip` together the results from the first loop and `bound_generic_params`. 

I don't understand how that works out, or at least, I don't see how it makes the code simpler.

Reply:
My bad, hadn't seen that you used `param` in the second field now.